src/datanode/src/config.rs

```diff
@@ -60,6 +60,14 @@ impl ObjectStoreConfig {
         }
     }
 
+    /// Returns true when it's a remote object storage such as AWS s3 etc.
+    pub fn is_object_storage(&self) -> bool {
+        matches!(
+            self,
+            Self::S3(_) | Self::Oss(_) | Self::Azblob(_) | Self::Gcs(_)
+        )
+    }
```

Comment:
Nit: how about only excluding `Self::File`? This is not likely to change compared to the OSS services.

Reply:
+1